### PR TITLE
feat: add ai bot template

### PR DESCRIPTION
**Pull request by swatDong**
Workitem: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/24551557
E2E TEST: N/A

Create new AI chat bot app:
New App

Create Bot

Run AI chat bot:
AI Bot

### CODE REVIEW

File: templates/js/ai-bot/src/app.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+const { MemoryStorage } = require("botbuilder");
+const path = require("path");
+const config = require("./config");
+
+const {
+  Application,
+  DefaultPromptManager,
+  OpenAIPlanner,
+  AzureOpenAIPlanner,
```
[the remark this answers is not in the document]
Comment (swatDong) [reply templates/js/ai-bot/src/app.js:9]: As designed, commented code in template.

File: templates/ts/ai-bot/src/app.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import { MemoryStorage } from "botbuilder";
+import * as path from "path";
+
+import {
+  Application,
+  AzureOpenAIPlanner,
+  DefaultPromptManager,
+  OpenAIPlanner,
+} from "@microsoft/teams-ai";
```
[the remark this answers is not in the document]
Comment (swatDong) [reply templates/ts/ai-bot/src/app.ts:9]: As designed, commented code in template.

File: templates/js/ai-bot/src/app.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  defaultModel: "gpt-3.5-turbo",
+  useSystemMessage: true,
+  logRequests: true,
+});
+// Uncomment the following lines to use Azure OpenAI
+/**
+const planner = new AzureOpenAIPlanner({
+  apiKey: config.azureOpenAIKey,
+  endpoint: config.azureOpenAIEndpoint,
+  defaultModel: "gpt-35-turbo",
```
Comment (kimizhu) [inline templates/js/ai-bot/src/app.js:24]: gpt-3.5-turbo or gpt-35-turbo?

File: templates/js/ai-bot/src/app.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    promptManager,
+    prompt: "chat",
+    history: {
+      assistantHistoryType: "text",
+    },
+  },
+});
+
+app.conversationUpdate("membersAdded", async (context) => {
+  await context.sendActivity("I am an AI assistant that helps people find information.");
```
Comment (kimizhu) [inline templates/js/ai-bot/src/app.js:46]: Is "How can I help you today?" better? :)

**Review by jayzhang — APPROVED** [review]

File: templates/js/ai-bot/src/app.js (hunk 4 above)
Comment (swatDong) [reply templates/js/ai-bot/src/app.js]: Updated

File: templates/js/ai-bot/src/app.js (hunk 3 above)
Comment (swatDong) [reply templates/js/ai-bot/src/app.js:24]: OpenAI uses `gpt-3.5-turbo` and Azure OpenAI uses `gpt-35-turbo` (https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models#gpt-35)

**Review by sffamily — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Review by kimizhu — APPROVED** [review]

**Pull request merged**